system/jlib/jptree.cpp

```diff
@@ -5734,19 +5737,25 @@ static void _toJSON(const IPropertyTree *tree, IIOStream &out, unsigned indent,
     {
         Linked<IPropertyTree> element = &sub->query();
         const char *name = element->queryName();
-        if (sub->next() && !repeatingElement && streq(name, sub->query().queryName()))
+        bool isBinary = element->isBinary(NULL);
+        bool match = false;
+        if (sub->next())
         {
-            if (flags & JSON_Format)
-                indent++;
-            writeJSONNameToStream(out, name, (flags & JSON_Format) ? indent : 0, delimit);
-            writeCharToStream(out, '[');
-            repeatingElement = true;
-            delimit = false;
+            match = (isBinary==sub->query().isBinary(NULL)) && streq(name, sub->query().queryName());
```

Comment:
Not that it would be common, but what happens if the same element is added multiple times, but sometimes binary and sometimes not?
I.e.

```
tree->addPropBin("abc",mb.length(), mb.length(), mb.toByteArray());
tree->addProp("abc", "xyz";
tree->addPropBin("abc",mb.length(), mb.length(), mb.toByteArray());
tree->addProp("abc", "xyz";
tree->addPropBin("abc",mb.length(), mb.length(), mb.toByteArray());
tree->addProp("abc", "xyz";
```
I would think the proper JSON would be 2 arrays {"$abc": ["<data>", "<data>", "<data>"], "abc": ["xyz", "xyz", "xyz"]}

But unless the binaries end up grouped together, would this current implementation end up in 6 single entries (or if some are grouped a variable number of entries)?

Reply:
Items will be grouped in JSON output as per your example, if they are consecutive. 
But don't think it should otherwise, because that would lose order which is normally preserved for matching property names.

Reply:
I understand wanting to preserve the order.  It technically would be invalid JSON unless you further mangled the names to make each unique.  Loading it back into a PropertyTree would still work though.

Reply:
Right I see, didn't realize it was invalid JSON, that's unfortunate.
It would be pain to preserve order in more convoluted way , but without it, it breaks existing ordering semantics.

Any suggestions?

Reply:
You could detect this case and simply further mangle the name.  I.e. something like:
```
{"a": {"b$": ["<data>", "<data>"], "b": ["abc"], "b$.2": ["<data>"], "b.2": ["abc"]}}
```
It's a complicated work around for a rare issue, but we could standardize on it in case similar issues came up in the future... always treat trailing ".n" as a continuation of a repeating element.

Certainly open to other variations of name mangling, or any other ideas.